engine/crates/runtime-local/src/fetch.rs

```diff
@@ -36,6 +47,8 @@ impl Fetcher for NativeFetcher {
         *response.extensions_mut() = extensions;
         *response.headers_mut() = headers;
 
+        response.extensions_mut().insert(info.finalize(status.as_u16()));
```

Comment:
I wouldn't pass it through `extensions` imo, why go through the AnyMap when you can return a tuple?